File: 0788-rotated-digits/0788-rotated-digits.py
class Solution:
    def rotatedDigits(self, n: int) -> int:
        ans = 0
        for i in range(1, n + 1, 1):
            if good(i):
                ans += 1
        
        return ans
        
def good(n):
    valid = False
    while n > 0:
        d = n % 10

        if d == 3 or d == 4 or d == 7:
            return False

        if d == 2 or d == 5 or d == 6 or d == 9:
            valid = True
        
        n //= 10

    return valid

# brute force
# time: O(nlog(n))
# space: O(1)

File: 0788-rotated-digits/0788-rotated-digits.go
func rotatedDigits(n int) int {
	ans := 0
	for i := 1; i <= n; i++ {
        if good(i) {
            ans++
        }
	}

    return ans
}

func good(x int) bool {
    valid := false
    for x > 0 {
        d := x % 10

        if d == 3 || d == 4 || d == 7 {
            return false
        }

        if d == 2 || d == 5 || d == 6 || d == 9 {
            valid = true
        }

        x /= 10
    }

    return valid
}

// brute force
// time: O(nlog(n))
// space: O(1)